lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -826,26 +838,6 @@ void ProxyRenderDelegate::_UpdateRenderTags()
         for (auto& id : rprimsToDirty) {
             changeTracker.MarkRprimDirty(id, HdChangeTracker::DirtyRenderTag);
         }
-
-        // Make a special pass over the removed render tags so that the objects
-        // can hide themselves.
-        _taskController->SetRenderTags(removedRenderTags);
-        _engine.Execute(_renderIndex.get(), &_dummyTasks);
-
-        // Set the new render tags correctly.The regular execute will cause
-        // any newly visible rprims will update and mark themselves visible.
-        TfTokenVector renderTags
-            = { HdRenderTagTokens->geometry }; // always draw geometry render tag purpose.
-        if (_proxyShapeData->DrawRenderPurpose()) {
-            renderTags.push_back(HdRenderTagTokens->render);
-        }
-        if (_proxyShapeData->DrawProxyPurpose()) {
-            renderTags.push_back(HdRenderTagTokens->proxy);
-        }
-        if (_proxyShapeData->DrawGuidePurpose()) {
-            renderTags.push_back(HdRenderTagTokens->guide);
-        }
-        _taskController->SetRenderTags(renderTags);
```

Comment:
Don't set the render tags to the current set of drawn tags. We leave them as all render tags, because we have to catch those cases where an rprim goes from a visible tag to a hidden tag.